src/renderer/renderer_elements.ts

```diff
@@ -860,6 +862,152 @@ export class ReturnBlock extends BasicBlock {
 
 }
 
+export class ConditionalRegion extends ControlFlowRegion {
+    
+    public static get CONDITION_SPACING(): number {
+        return 3 * SDFV.LINEHEIGHT;
+    }
+
+    public static get LOOP_STATEMENT_FONT(): string {
+        return (SDFV.DEFAULT_CANVAS_FONTSIZE * 1.5).toString() +
+            'px sans-serif';
+    }
+
+    public branches: ([{string_data: string, language: string}, region: ControlFlowRegion])[] = []
+
+    public draw(
+        renderer: SDFGRenderer, ctx: CanvasRenderingContext2D,
+        _mousepos?: Point2D
+    ): void {
+        const topleft = this.topleft();
+        const visibleRect = renderer.get_visible_rect();
+
+        let clamped;
+        if (visibleRect) {
+            clamped = {
+                x: Math.max(topleft.x, visibleRect.x),
+                y: Math.max(topleft.y, visibleRect.y),
+                x2: Math.min(
+                    topleft.x + this.width, visibleRect.x + visibleRect.w
+                ),
+                y2: Math.min(
+                    topleft.y + this.height, visibleRect.y + visibleRect.h
+                ),
+                w: 0,
+                h: 0,
+            };
+        } else {
+            clamped = {
+                x: topleft.x,
+                y: topleft.y,
+                x2: topleft.x + this.width,
+                y2: topleft.y + this.height,
+                w: 0,
+                h: 0,
+            };
+        }
+        clamped.w = clamped.x2 - clamped.x;
+        clamped.h = clamped.y2 - clamped.y;
+        if (!renderer.viewportOnly) {
+            clamped = {
+                x: topleft.x,
+                y: topleft.y,
+                x2: 0,
+                y2: 0,
+                w: this.width,
+                h: this.height,
+            };
+        }
+
+        // Draw the loop background below everything and stroke the border.
+        ctx.fillStyle = this.getCssProperty(
+            renderer, '--conditional-background-color'
+        );
+        ctx.strokeStyle = this.getCssProperty(
+            renderer, '--conditional-foreground-color'
+        );
+        ctx.fillRect(clamped.x, clamped.y, clamped.w, clamped.h);
+
+        // Only draw line if close enough.
+        const ppp = renderer.get_canvas_manager()?.points_per_pixel();
+        if (!renderer.adaptiveHiding || (ppp && ppp < SDFV.NODE_LOD))
+            ctx.strokeRect(clamped.x, clamped.y, clamped.w, clamped.h);
+
+        ctx.fillStyle = this.getCssProperty(
+            renderer, '--loop-foreground-color'
+        );
+
+        let topSpacing = LoopRegion.META_LABEL_MARGIN;
```

Comment:
Probably makes sense to refactor `META_LABEL_MARGIN` to `ControlFlowRegion` if now multiple subclasses use it